architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,276 @@
+   "hostName": "bicep-de-6b86cc859f-tt2r8",
+   "traceId": "a0d750fc4e5d35bd1c615d87c1261f72",
+   "spanId": "5b624ecddd0b19c2",
+   "HttpMethod": "GET",
+   "Uri": "https://ucp.radius-system/apis/api.ucp.dev/v1alpha3/planes/radius/local/resourceGroups/default/providers/Applications.Core/containers/demo?api-version=2023-10-01-preview"
+   },
+   {
+   "timestamp": "2024-11-03T14:06:54.684Z",
+   "severity": "INFO",
+   "message": "Sending HTTP request GET https://ucp.radius-system/apis/api.ucp.dev/v1alpha3/planes/radius/local/resourceGroups/default/providers/Applications.Core/containers/demo?api-version=2023-10-01-preview",
+   "name": "RequestStart",
+   "scope": "System.Net.Http.HttpClient.local.ClientHandler",
+   "serviceName": "deployment.engine",
+   "version": "0.38\u002B1b2449770894506793f865dc7ecc9ed8e6df0d93",
+   "hostName": "bicep-de-6b86cc859f-tt2r8",
+   "traceId": "a0d750fc4e5d35bd1c615d87c1261f72",
+   "spanId": "5b624ecddd0b19c2",
+   "HttpMethod": "GET",
+   "Uri": "https://ucp.radius-system/apis/api.ucp.dev/v1alpha3/planes/radius/local/resourceGroups/default/providers/Applications.Core/containers/demo?api-version=2023-10-01-preview"
+   }
+   ```
+
+2. **Health Check Probes**: Kubernetes itself can act as a client by performing health and readiness checks on the Universal Control Plane.
+
+3. **Metrics Scrapers**: If metrics are enabled, Prometheus or other monitoring tools can scrape metrics from the Universal Control Plane.
+
+## Trust Boundaries
+
+### Trust Model of UCP Clients
+
+The clients of the UCP include other Radius components like Applications RP, CLI, Controller, Dashboard, and Deployment Engine. Most of these clients reside within the same Kubernetes cluster and namespace as the UCP, except for the Radius CLI, which operates outside of the Kubernetes cluster and, therefore, outside of the namespace.
+
+For clients that are in the same cluster and namespace as the UCP component, we rely on Kubernetes to implement the necessary security measures and establish the required trust boundary. For external clients--meaning clients that are outside of this trust boundary--we rely on the host machine's trust boundary and assume that it is not compromised. In such cases, proper authentication and authorization are essential to maintain security.
+
+### Trust Model of Internal Resource providers
+
+Currently, our only resource provider is the Applications RP. As mentioned above, the Applications RP lives in the same Kubernetes cluster and namespace as the UCP. Therefore, the trust boundaries are defined by the Kubernetes cluster and the namespace within that cluster.
+
+### Trust Model of External Resource Manages (Azure and AWS)
+
+The UCP routes user requests to Azure and AWS whenever a resource from those providers is requested. In these cases, we rely on Azure and AWS to establish the trust boundary and operate under the assumption that they are secure and trustworthy.
```

Comment:
We should talk about how this is secured. I think we can make a strong assumption by discussing authentication.

Reply:
Should we talk about the credentials that the user provides us so that we can authenticate to these cloud providers?

Reply:
Updated this part:

```
The UCP routes user requests to Azure and AWS when resources from these providers are needed. We rely on Azure and AWS to establish the trust boundary and assume they are secure and trustworthy. We use the credentials provided by the user to authenticate with these cloud providers and then trigger the necessary actions.

Communication with external resource managers is done through their public HTTPS APIs using standard authentication mechanisms. Therefore, we assume that these external resource managers are authentic and that our requests are subject to their authorization models.
```